plugins/scanners/dos/src/main/kotlin/DosScanner.kt

```diff
@@ -105,7 +105,11 @@ class DosScanner internal constructor(
                 return@runBlocking null
             }
 
-            val packages = context.coveredPackages.getDosPackages(provenance)
+            val packages = mutableListOf<PackageInfo>()
+
+            nestedProvenance.allProvenances.forEach {
+                packages.addAll(context.coveredPackages.getDosPackages(it))
+            }
```

Comment:
This can be simplified like so:

            val packages = nestedProvenance.allProvenances.flatMap {
                context.coveredPackages.getDosPackages(it)
            }